experiments/tf_trainer/common/tfrecord_input.py

```diff
@@ -8,43 +8,34 @@
 
 import tensorflow as tf
 from tf_trainer.common import dataset_input
+from tf_trainer.common import tfrecord_simple
 from tf_trainer.common import types
+
 from typing import Callable, Dict, List
 
 
-class TFRecordInput(dataset_input.DatasetInput):
+class TFRecordInput(tfrecord_simple.TFSimpleRecordInput):
   """TFRecord based DatasetInput.
 
   Handles parsing of TF Examples.
   """
 
   def __init__(self,
-               train_path: str,
-               validate_path: str,
-               text_feature: str,
-               labels: Dict[str, tf.DType],
                train_preprocess_fn: Callable[[str], List[str]],
                batch_size: int = 64,
-               round_labels: bool = True,
                num_prefetch: int = 5,
                max_seq_len: int = 30000) -> None:
-    self._train_path = train_path
-    self._validate_path = validate_path
-    self._text_feature = text_feature
-    self._labels = labels
-    self._batch_size = batch_size
+    super().__init__(batch_size, num_prefetch)
     self._train_preprocess_fn = train_preprocess_fn
-    self._round_labels = round_labels
-    self._num_prefetch = num_prefetch
     self._max_seq_len = max_seq_len
 
-  def train_input_fn(self) -> types.FeatureAndLabelTensors:
-    """input_fn for TF Estimators for training set."""
-    return self._input_fn_from_file(self._train_path)
+  def tokens_feature(self) -> str:
+    """Returns the name of the feature containing integer tokens.
 
-  def validate_input_fn(self) -> types.FeatureAndLabelTensors:
-    """input_fn for TF Estimators for validation set."""
-    return self._input_fn_from_file(self._validate_path)
+    Note that currently this code overwrites the input text feature
```

Comment:
Yes, I want to fix this and tried to create a new 'tokens' feature that we use instead of editing the content (and type!) of the feature. But: it's hard coded in the keras code to use the 'comment_text' feature
https://github.com/conversationai/conversationai-models/blob/master/experiments/tf_trainer/keras_cnn/model.py#L67
one step at a time?